src/main/java/org/hydev/mcpm/client/database/SuperLocalPluginTracker.java

```diff
@@ -0,0 +1,619 @@
+     *
+     * @return List of plugin names
+     */
+    public List<String> listManuallyInstalled() {
+        ArrayList<PluginTrackerModel> currentList = readCsv();
+
+        ArrayList<String> newList = new ArrayList<>();
+
+        for (PluginTrackerModel pluginTrackerModel : currentList) {
+            if (pluginTrackerModel.isManual) {
+                newList.add(pluginTrackerModel.name);
+            }
+        }
+
+        return newList;
+    }
+
+    /**
+     * Get a list of automatically installed plugin dependencies that are no longer
+     * required
+     *
+     * @return List of plugin names
+     */
+    public List<String> listOrphanPlugins(boolean considerSoftDependencies) {
+
+        List<String> orphanPlugins = new ArrayList<>();
+        List<String> manuallyinstalledPlugins = listManuallyInstalled();
+        List<String> requiredDependencies = new ArrayList<>();
+
+        // Get all the dependencies of the manually installed plugins
+        for (String name : manuallyinstalledPlugins) {
+            try {
+                // Find the pluginYml file of the plugin with name name from the plugin
+                // directory
+
+                File pluginYmlPath = getPluginFile(name);
+                PluginYml p = readMeta(pluginYmlPath);
+                if (p == null)
+                    continue;
+
+                // String pluginYmlPath = pluginDirectory + "/" + name + "/plugin.yml";
+                // PluginYml currPlugin = readMeta(new File(pluginYmlPath));
+                // Add the dependencies of the plugin to the list of required dependencies
+                if (p.depend() != null)
+                    requiredDependencies.addAll(p.depend());
+
+                // If considerSoftDependencies is true, add the soft dependencies to the list of
+                // required dependencies
+                if (considerSoftDependencies) {
+                    if (p.softdepend() != null)
+                        requiredDependencies.addAll(p.softdepend());
+                }
+
+            } catch (Exception e) {
+                System.out.println("Error getting dependencies of " + name);
+            }
+        }
+
+        // Get the difference between the set of manually installed plugins,
+        // the set of required dependencies, and the set of all installed plugins.
+        List<PluginYml> installedPluginsYml = listInstalled();
+
+        // Create a list of all installed plugin names in string format from the
+        // installedPluginsYML list
+        List<String> installedPlugins = new ArrayList<>();
+        for (PluginYml plugin : installedPluginsYml) {
+            installedPlugins.add(plugin.name());
+        }
+
+        // Get the difference between the set of manually installed plugins,
+        // the set of required dependencies, and the set of all installed plugins
+        orphanPlugins = installedPlugins;
+        orphanPlugins.removeAll(requiredDependencies);
```

Comment:
if you're going to be removing, you might as well make requiredDependencies and manuallyinstalledPlugins sets